### DISCUSSION TITLE
utf-16 encoding

### DISCUSSION THREAD

**Discussion by prabhu-yu** [micropython/micropython#13481] [Libraries & Drivers]
I want to encode a string into utf-16 bytes.
In Python3, I know, but in Micropython, I do not.
The following example code works in Python3(Ubuntu Linux) but not in MicroPython (ESP32/RPI Pico).
Some please let me know how to achieve this.
```
s = '8'
t = s.encode('utf_16_be')
for i in t:
    print(i)
```

```
#Correct Output in Ubuntu
#python3 2.py
0
56
```

```
#Wrong output in MicroPython
>>> %Run -c $EDITOR_CONTENT
56
```

**Comment by cwalther** [comment]
The only encoding supported by MicroPython (as far as I know), for size reasons, is UTF-8. It simply ignores the encoding argument to any functions that take one and always uses UTF-8. You will have to implement the conversion yourself (or find a library that has already done it).

**Comment by scruss** [comment]
I implemented a very crude UTF-16BE encoder in my [micropython-SYN6988](https://github.com/scruss/micropython-SYN6988) speech synthesizer project. It might do what you need:

```python3
def fauxutf16(s, badchar=" "):
    # encodes string s as UTF-16BE, two bytes per char, no BOM
    # eg: "hello" => b'\x00h\x00e\x00l\x00l\x00o'
    # char codes > 65535 encoded as badchar
    # MicroPython can only encode to UTF-8 bytes, so we need this
    buf = bytearray(len(s) * 2)  # preallocate empty buffer
    for i, c in enumerate(s):
        n = ord(c)
        if n > 65535:
            n = ord(badchar)
        buf[2 * i] = n // 256
        buf[2 * i + 1] = n % 256
    return bytes(buf)

s = '8'
t = fauxutf16(s)
for i in t:
    print(i)
```

Output from MicroPython and Python is:
```
0
56
```
